server/routes/api.js

```diff
@@ -536,17 +537,52 @@ router.get('/capcodes/agency/:id', isLoggedIn, function(req, res, next) {
 // POST calls below
 //
 //////////////////////////////////
+
+// dupe init
+var msgBuffer = [];
+
 router.post('/messages', isLoggedIn, function(req, res, next) {
   nconf.load();
   if (req.body.address && req.body.message) {
     var filterDupes = nconf.get('messages:duplicateFiltering');
     var dupeLimit = nconf.get('messages:duplicateLimit') || 0; // default 0
     var dupeTime = nconf.get('messages:duplicateTime') || 0; // default 0
     var pdwMode = nconf.get('messages:pdwMode');
-    var adminShow = nconf.get('messages:adminShow')
+    var adminShow = nconf.get('messages:adminShow');
     var data = req.body;
         data.pluginData = {};
 
+    if (filterDupes) {
+      // this is a bad solution and tech debt that will bite us in the ass if we ever go HA, but that's a problem for future me and that guy's a dick
+      var datetime = data.datetime || 1;
+      var timeDiff = datetime - dupeTime;
+      // if duplicate filtering is enabled, we want to populate the message buffer and check for duplicates within the limits
+      var matches = _.where(msgBuffer, {message: data.message, address: data.address});
+      if (matches.length > 0) {
+        if (dupeTime != 0) {
+          // search the matching messages and see if any match the time constrain
+          var timeFind = _.find(matches, function(msg){ return msg.datetime > timeDiff; });
+          if (timeFind) {
+            res.status(200);
+            return res.send('Ignoring duplicate');
```

Comment:
Should have an info logger here too 
logger.main.info(util.format('Ignoring duplicate: %o', message));